hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -182,9 +187,10 @@ public HoodieCompactedLogRecordScanner(FileSystem fs, String basePath, List<Stri
    * the log records since the base data is merged on previous compaction
    */
   private Map<String, HoodieRecord<? extends HoodieRecordPayload>> loadRecordsFromBlock(
-      HoodieAvroDataBlock dataBlock) {
-    Map<String, HoodieRecord<? extends HoodieRecordPayload>> recordsFromLastBlock = Maps
-        .newHashMap();
+      HoodieAvroDataBlock dataBlock) throws IOException {
```

Comment:
I think we should simplify this more by assuming that a `HoodieAvroDataBlock` can be held in memory.. i.e during write we batch and create multiple data blocks..

Reply:
You mean just use the old implementation of Maps.newHashMap() ? The problem is we only write `1` data block during a write batch so the size of this can be fairly large given we have the other `records` which hold all compacted records.

Reply:
@vinothchandar here, I think you're referring to just just Maps.newHashMap instead of spillable ? With the new logformat changes and lazy reading of data, this should work.